pkg/autoscaling/handler.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package autoscaling
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/pd/v4/server"
+	"github.com/unrolled/render"
+)
+
+// Strategy within a HTTP request provides rules and resources to help make decision for auto scaling.
+type Strategy struct {
+	Rules     []*Rule     `json:"rules"`
+	Resources []*Resource `json:"resources"`
+}
+
+// Rule is a set of constraints for a kind of component.
+type Rule struct {
+	Component               string       `json:"component"`
+	CPURule                 *CPURule     `json:"cpu_rule"`
+	StorageRule             *StorageRule `json:"storage_rule"`
```

Comment:
Do we need to add `omitempty` tag here?

Reply:
I think we need to keep at least one rule. Maybe StorageRule can be omitted but CPURule can not?

Reply:
I think we can make both rule as `omitempty` here while we could verify rule count shouldn't be less than 1 in code logic.